ansible_ai_connect/ai/api/model_pipelines/http/pipelines.py

```diff
@@ -107,3 +117,55 @@ def self_test(self) -> HealthCheckSummary:
 
     def infer_from_parameters(self, api_key, model_id, context, prompt, suggestion_id=None):
         raise NotImplementedError
+
+
+@Register(api_type="http")
+class HttpChatBotPipeline(HttpMetaData, ModelPipelineChatBot[HttpConfiguration]):
+
+    def __init__(self, config: HttpConfiguration):
+        super().__init__(config=config)
+
+    def invoke(self, params: ChatBotParameters) -> ChatBotResponse:
+        query = params.query
+        conversation_id = params.conversation_id
+        provider = params.provider
+        model_id = params.model_id
+
+        data = {
+            "query": query,
+            "model": model_id,
+            "provider": provider,
+        }
+        if conversation_id:
+            data["conversation_id"] = str(conversation_id)
+
+        response = requests.post(
+            self.config.inference_url + "/v1/query",
+            headers=self.headers,
+            json=data,
+            timeout=self.timeout(1),
```

Comment:
Each _service_/_feature_ can have its own independent configuration.

I set `timeout` to `ANSIBLE_AI_MODEL_MESH_API_TIMEOUT` by default following the others.

However I've now removed the timeout for ChatBot all together.

The environment variables are now obsolete; they:-

- Are only used to build the JSON block `ANSIBLE_AI_MODEL_MESH_CONFIG` which drives everything.
- Allow us to run the _new code_ without having to change AWS Secret Manager or our tests (for now).
- Will be deleted with https://issues.redhat.com/browse/AAP-36732

Reply:
ChatBot can _still have_ a timeout (later, if needed).. so this bit of the code stays unchanged.

I call `self.timeout(1)` so that whatever timeout has been configured is used.

The same _code_ is in a super-class used also by Completions. Completions increases the timeout based on the number of tasks in a multi-prompt. You _could_ (if you ever wanted) change this to timeout based on the number of messages in the chat history, for example.